src/main/java/com/google/devtools/build/lib/rules/genrule/GenRuleBase.java

```diff
@@ -115,28 +149,35 @@ public ConfiguredTarget create(RuleContext ruleContext)
       return null;
     }
 
-    String baseCommand = ruleContext.attributes().get("cmd", Type.STRING);
+    CommandType cmdType = cmdTypeAndAttr.first;
+    String cmdAttr = cmdTypeAndAttr.second;
+    boolean expandToWindowsPath = cmdType != CommandType.BASH;
+
+    String baseCommand = ruleContext.attributes().get(cmdAttr, Type.STRING);
+
     // Expand template variables and functions.
     ImmutableList.Builder<MakeVariableSupplier> makeVariableSuppliers =
         new ImmutableList.Builder<>();
     CommandResolverContext commandResolverContext =
         new CommandResolverContext(
-            ruleContext, resolvedSrcs, filesToBuild, makeVariableSuppliers.build());
+            ruleContext, resolvedSrcs, filesToBuild, makeVariableSuppliers.build(), expandToWindowsPath);
     String command =
         ruleContext
             .getExpander(commandResolverContext)
-            .withExecLocations(commandHelper.getLabelMap())
-            .expand("cmd", baseCommand);
+            .withExecLocations(commandHelper.getLabelMap(), expandToWindowsPath)
+            .expand(cmdAttr, baseCommand);
 
     // Heuristically expand things that look like labels.
     if (ruleContext.attributes().get("heuristic_label_expansion", Type.BOOLEAN)) {
       command = commandHelper.expandLabelsHeuristically(command);
     }
 
-    // Add the genrule environment setup script before the actual shell command.
-    command = String.format("source %s; %s",
-        ruleContext.getPrerequisiteArtifact("$genrule_setup", Mode.HOST).getExecPath(),
-        command);
+    if (cmdType == CommandType.BASH) {
+      // Add the genrule environment setup script before the actual shell command.
+      command = String.format("source %s; %s",
+          ruleContext.getPrerequisiteArtifact("$genrule_setup", Mode.HOST).getExecPath(),
```

Comment:
Do you plan to allow the same for cmd_exe / ps commands?

Reply:
Nope, we can setup the environment without a setup script in those two.